src/ir/module-splitting.cpp

```diff
@@ -409,6 +434,21 @@ void ModuleSplitter::thunkExportedSecondaryFunctions() {
   }
 }
 
+Expression* ModuleSplitter::maybeLoadSecondary(Builder& builder,
+                                               Expression* callIndirect) {
+  if (config.jspi) {
+    // Check if the secondary module is loaded and if it isn't, call the
+    // function to load it.
```

Comment:
Does this depend on the secondary module being loaded synchronously? Or this this import JSPIed such that execution suspends until it asynchronously resolves? This stuff is so mind bending...

Reply:
`load_secondary_module` will be a JSPI function so it will suspend until the secondary module is loaded. Any exported wasm function that leads to this code path will also need a JSPI wrapper. If the user messes this up there will be a runtime error.

Reply:
I see, so it's up to the user to manually choose which exports to JSPI-ify? Can we add an option to trap or otherwise report an error if we ever get to one of these loading sites from a non-JSPI-ified entry point?

Reply:
> I see, so it's up to the user to manually choose which exports to JSPI-ify?

Yeah, we could do the automatic route, but then you may have exports that switch between sync and async.

> Can we add an option to trap or otherwise report an error if we ever get to one of these loading sites from a non-JSPI-ified entry point?

We could, though v8 will already reports this with an "invalid suspender error". Do we want to add our own trap as well?

Reply:
Ah, unless we can provide a better error message, I guess there's nothing more for us to do.